### PR TITLE
Accessibility docs

### PR DESCRIPTION
**Pull request by jamespeilow**
# What are you trying to accomplish with this PR?

Added an Accessibility section to the coding guidelines

**Task:** CANVAS-479

# Checklist
For contributors (tick with ☑):

### Priority
* ☐ Chrome, desktop
* ☐ Safari, iOS
### Supplemental
* ☐ Firefox, desktop
* ☐ Safari, desktop
* ☐ Chrome, Android

## Accessibility
* ☐ New elements have been tested in Lighthouse and/or an extension like [Axe](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd)
* ☐ Elements can be navigated with keyboard controls
* ☐ Elements are labelled and screen-reader compatible

## Coding
* ☐ Follows [coding guidelines](https://github.com/we-make-websites/wmw-coding-guidelines)
* ☐ There are no eslint or stylelint linting errors
* ☐ The code is logical and commented
* ☐ Async components have a static loading state
* ☐ Documentation has been updated to reflect the changes, if applicable
* ☐ There are no hardcoded string values in Liquid
* ☐ All new Liquid locale strings are added to each _shopify/locales/_ file
* ☐ This work is inline with WCAG 2.1 AA standards

### CODE REVIEW

File: accessibility/README.md [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+
+<a href="/sale">
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg" alt="Get 10% off">
+</a>
+```
+
+* When using `<button>` or `<a>` elements containing SVG elements, ensure that accessible text is also included
+  * If the text should not be visible, the `visually-hidden` class can be used
+* If a `<button>` or `<a>` contains a `<img>` element - ensure that there is either appropriate `alt` text on the image, or add a supporting `visually-hidden` span with appropriate text
```
Comment (georgia-wmw) [inline accessibility/README.md:346]: Would it be possible to add a note that if `visually-hidden` text is used, the `alt` should be empty? Otherwise screenreaders will read out possibly duplicate or irrelevant content

File: accessibility/README.md [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+* Ensure that the element referenced by `aria-labelledby` exists on the page.
+  * If the referenced element is missing, screen readers will still read out the the other text content contained in the element, but users may be missing out on the additional context that the referenced element might be adding
+* When using `aria-labelledby` in a component, consider whether the element you are referencing existing inside the component's template code.
+  * If the element you are targeting with `aria-labelledby` is stored in another component or snippet, you may not be able to guarantee that it will always be on the page.
+  * In this case, opt to use `<span class="visually-hidden">` or `aria-label` instead.
+* Additional documentation: [MDN - aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby)
+
+#### `aria-label`
+* `aria-label` is another approach for labelling elements, and can be used when there is no visible element in the DOM that can be used directly as a label or referenced with `aria-labelledby`
+* Examples of using this are similar to adding a `<span class="visually-hidden">` element
```
Comment (georgia-wmw) [inline accessibility/README.md:385]: Would it be worth adding some context why we're using `visually-hidden` text in the above examples?

I think I generally use them because `aria-label` has had a rocky history with screenreader support ([see this blog post from 2020](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/)), and printed text is always read out (unless `aria-hidden` or `display: none` is present - but I don't know if that's actually the reason.

File: accessibility/README.md [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+    namespace: {
+      type: String,
+      default: 'default'
+      required: true,
+    },
+    
+    options: {
+      type: Array,
+      default: () => [],
+
```
Comment (georgia-wmw) [inline accessibility/README.md]: How nitpicky should we be with code examples 👀 
Looks like there's an extra newline here that can be removed

**Review by georgia-wmw — CHANGES_REQUESTED** [review]
🎉 

I've got a couple of questions on if we could add or flesh out some things but this information is outlined really well!

File: accessibility/README.md (hunk 1 above)
Comment (jamespeilow) [reply accessibility/README.md:346]: Good point - I've added a line about this

File: accessibility/README.md (hunk 2 above)
Comment (jamespeilow) [reply accessibility/README.md:385]: This is good to know - I've always preferred using `visually-hidden` over `aria-label` - so its good to have some reasons why to back this up.
I've added some notes based on that blog post and linked to it directly

File: accessibility/README.md (hunk 3 above)
Comment (jamespeilow) [reply accessibility/README.md]: Good spot - Its a little harder to pick up code example styleguide issues when I'm working just in markdown.

Fixed this 👍

**Comment by jamespeilow** [comment]
@georgia-wmw - I've just actioned your PR feedback and replied to your comments, so this is now ready to re-review

File: accessibility/README.md (hunk 2 above)
Comment (georgia-wmw) [reply accessibility/README.md:385]: Oooh I didn't even think about translation! Nice!

**Review by georgia-wmw — APPROVED** [review]
Looks good to me! Definitely exciting to lay out some standards for accessibility 🎉

File: accessibility/README.md [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+<div
+  class="dropdown-menu__input"
+  @click="toggleMenu"
+>
+  <span v-text="label">
+  <icon-chevron-down />
+</div>
+
+<responsive-image
+  @click="handleImageClick"
```
Comment (craigsbaldwin) [inline accessibility/README.md]: Single attributes should be on the same line.

File: accessibility/README.md [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+  role="button"
+  tabindex="0"
+  @click="handleClick"
+  @keydown="handleKeyDown"
+>
+  Click me
+</div>
+
+// Good
+<button @click=:handleClick>
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
<button @click="handleClick">
```
Right? At least to be consistent with your previous 'Good' example.

File: accessibility/README.md [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+### Don't use `<a>` elements where a `<button>` would be better
+
+#### Don't
+
+```vue
+<a @click="toggleItem">
+  Click here
+</a>
+
+<a href="#" @click.prevent="doSomething">
```
Comment (craigsbaldwin) [inline accessibility/README.md]: Multiple attributes should be split onto multiple lines.

File: accessibility/README.md [hunk 7]
```diff
@@ -0,0 +1,495 @@
...
+  Click here
+</button>
+
+<button @click.prevent="doSomething">
+  Do something
+</button>
+```
+
+* If an element is not linking somewhere, it's better to use a `<button>` element instead
+  * `<a>` tags should not be used without a `href` attribute or with blank or javascript `href` attribute
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
* `<a>` tags should not be used without a `href` attribute or with blank or javascript `href` attribute
```
I'd say this bullet is a 'sub' bullet of the preceding one (it's not really related) so can be outdented.

File: accessibility/README.md [hunk 8]
```diff
@@ -0,0 +1,495 @@
...
+  :key="item.id"
+  @click="filter(item.value)"
+>
+  <input
+    :id="item.id"
+    :checked="item.active"
+    type="checkbox"
+    :value="item.value"
+  >
+  <label
```
Comment (craigsbaldwin) [inline accessibility/README.md:154]: Nitpick: Newline between multi-line elements.

File: accessibility/README.md [hunk 9]
```diff
@@ -0,0 +1,495 @@
...
+>
+  <input
+    :id="item.id"
+    :checked="item.active"
+    type="checkbox"
+    :value="item.value" 
+    @change="filter($event.target.value)"
+  >
+
+  <label :for="item.id" v-text="item.label" />
```
Comment (craigsbaldwin) [inline accessibility/README.md]: Each attribute gets its own line.

File: accessibility/README.md [hunk 10]
```diff
@@ -0,0 +1,495 @@
...
+#### Do
+
+```vue
+<li
+  v-for="item in items"
+  :key="item.id"
+>
+  <input
+    :id="item.id"
+    :checked="item.active"
```
Comment (craigsbaldwin) [inline accessibility/README.md:170]: Possibly overcomplicates the example but `v-model` would be my preference for this sort of code.

File: accessibility/README.md [hunk 11]
```diff
@@ -0,0 +1,495 @@
...
+* [Focus Trap](#focus-trap)
+
+### Test keyboard focus
+* Ensure that interactive elements can be focussed and activated using a keyboard
+* Ensure that functionality that supports mouse interactions (click, mousedown, mouseleave etc) have suitable keyboard-accessible alternatives
+
+### Test keyboard tab order
+* Ensure that keyboard focus order is correct
+  * The order that items are focussed when tabbing should match the visual order of the elements on the page
+  * Try an ensure that the order elements in the DOM match the visual order on the site
```
Comment (craigsbaldwin) [inline accessibility/README.md]: Typo.
Try **and** ensure?
Try **to** ensure?

File: accessibility/README.md [hunk 12]
```diff
@@ -0,0 +1,495 @@
...
+* Exceptions can be made here when the order changes responsively, e.g. when the order is correct on mobile screen sizes 
+
+### Focus Trap
+
+* Use a focus trap when opening overlays so that a keyboard user is only able to tab between elements inside the overlay
+* Ensure that the focus trap is cleared when an overlay is closed
+* Ensure that the focus trap is updated when the overlay state changes
+  * Example: if you have a multi-tiered navigation, ensure that the focus trap updates as the user navigates between tiers
+  * Example: if you have a search overlay and the results update as you type, ensure that focus trap updates as the content changes
+* Focus trapping can be achieved in Canvas projects with the `focusTrap` and `previousElement` helpers in the `accessibility.js` store - see [Canvas Gitbook - accessibility.js](https://we-make-websites.gitbook.io/canvas/features/vuex-stores/accessibility.js)
```
Comment (craigsbaldwin) [inline accessibility/README.md]: Frame 3 documentation is kept separate in the _frame/_ folder. By default this documentation assumes you're talking about a Canvas project.

The fact that most of the documentation is written in Vue suggests that Frame 3 could be ignored here.

File: accessibility/README.md [hunk 13]
```diff
@@ -0,0 +1,495 @@
...
+
+### Ensure that `input` elements have corresponding `label` elements
+
+#### Don't
+
+```html
+<input
+  id="newsletter-email"
+  name="email" 
+  type="email"
```
Comment (craigsbaldwin) [inline accessibility/README.md]: Nitpick: `placeholder` comes before `type`.

File: accessibility/README.md [hunk 14]
```diff
@@ -0,0 +1,495 @@
...
+
+<input
+  id="newsletter-email"
+  name="email" 
+  type="email"
+  placeholder="Enter your email"
+>
+```
+
+* Ensuring that you have that an `input` has a corresponding `label` element helps to describe the purpose and can provide instructions for the input for users
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
* Ensuring that your `input` has a corresponding `label` element helps to describe the purpose of the input and can provide instructions for it for all users
```

File: accessibility/README.md [hunk 15]
```diff
@@ -0,0 +1,495 @@
...
+* A `label` should be used even when the input has a `placeholder` attribute.
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
* A `label` should be used even when the input has a `placeholder` attribute
```
Nitpick: Bullet points shouldn't end in a full stop.

File: accessibility/README.md [hunk 16]
```diff
@@ -0,0 +1,495 @@
...
+  * Assistive technology, such as screen-readers do not treat placeholder text as labels, so a `placeholder` should be used alongside a label, not instead of a label
+* In some cases, a site's design may show inputs without visible labels. Sometimes this may be because other elements on the page give the input some context - for example a search text input placed directly next to a "Search" button.
+  * If a input does not have a visible label, a `<label>` element should still be added for the benefit of screen readers
+  * A label can be hidden visually by adding the `visually-hidden` class to the `label`
+  ```html
+  <!-- Label will be available to assistive technology, but visually hidden -->
+  <label class="visually-hidden" for="search">
```
Comment (craigsbaldwin) [inline accessibility/README.md]: Attributes, single lines, etc.

Not going to point out further cases, please update all instances.

File: accessibility/README.md [hunk 17]
```diff
@@ -0,0 +1,495 @@
...
+  name="email" 
+  type="email"
+  placeholder="Enter your email"
+>
+```
+
+* Ensuring that you have that an `input` has a corresponding `label` element helps to describe the purpose and can provide instructions for the input for users
+* A `label` should be used even when the input has a `placeholder` attribute.
+  * Assistive technology, such as screen-readers do not treat placeholder text as labels, so a `placeholder` should be used alongside a label, not instead of a label
+* In some cases, a site's design may show inputs without visible labels. Sometimes this may be because other elements on the page give the input some context - for example a search text input placed directly next to a "Search" button.
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
* In some cases, a site's design may show inputs without visible labels, sometimes this may be because other elements on the page give the input some context - for example a search text input placed directly next to a 'Search' button
```
All sorts of nitpicks.

File: accessibility/README.md [hunk 18]
```diff
@@ -0,0 +1,495 @@
...
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg" alt="Get 10% off">
+</a>
+```
+
+* When using `<button>` or `<a>` elements containing SVG elements, ensure that accessible text is also included
+  * If the text should not be visible, the `visually-hidden` class can be used
+* If a `<button>` or `<a>` contains a `<img>` element - ensure that there is either appropriate `alt` text on the image, or add a supporting `visually-hidden` span with appropriate text.
+  * Avoid using both `alt` and `visually-hidden` text for the same element, otherwise screen readers will read out possibly duplicate or irrelevant content.
+  * Setting an empty `alt` text is fine when using `visually-hidden` text - see [Alt Text](#alt-text) for more details.
```
Comment (craigsbaldwin) [inline accessibility/README.md]: @georgia-wmw @jamespeilow Do we want to mention `role="presentation"`? I seem to remember aXe used to flag that an image that's purely decorative should have the `role` attribute.

Though that might be out-of-date now.

File: accessibility/README.md [hunk 19]
```diff
@@ -0,0 +1,495 @@
...
+```html
+<section aria-labelledby="template-title">
+  <h1 id="template-title">
+    Title for the section
+  </h1>
+
+  <!-- other content -->
+</section>
+```
+* Ensure that the element referenced by `aria-labelledby` exists on the page.
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
* Ensure that the element referenced by `aria-labelledby` exists on the page
```

File: accessibility/README.md [hunk 20]
```diff
@@ -0,0 +1,495 @@
...
+* The `aria-labelledby` and `aria-label` attributes provide another method for adding labels to elements
+* These are alternatives to using a `visually-hidden` element showcased in the examples above
+
+#### `aria-labelledby`
+
+* `aria-labelledby` can be used to label an element based on the text content of another existing element by it's unique `id`
+* If a heading is present inside a region, this could be a good candidate for `aria-labelledby`
+
+```html
+<section aria-labelledby="template-title">
```
Comment (craigsbaldwin) [inline accessibility/README.md:361]: Might want to flag why `section` elements should be labelled in the first place.

I'm still doing it because I think aXe once flagged that all `section` elements should be labelled.

File: accessibility/README.md [hunk 21]
```diff
@@ -0,0 +1,495 @@
...
+    Title for the section
+  </h1>
+
+  <!-- other content -->
+</section>
+```
+* Ensure that the element referenced by `aria-labelledby` exists on the page.
+  * If the referenced element is missing, screen readers will still read out the the other text content contained in the element, but users may be missing out on the additional context that the referenced element might be adding
+* When using `aria-labelledby` in a component, consider whether the element you are referencing existing inside the component's template code.
+  * If the element you are targeting with `aria-labelledby` is stored in another component or snippet, you may not be able to guarantee that it will always be on the page.
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
  * If the element you are targeting with `aria-labelledby` is stored in another component or snippet, you may not be able to guarantee that it will always be on the page
```

File: accessibility/README.md [hunk 22]
```diff
@@ -0,0 +1,495 @@
...
+
+<a href="/account" aria-label="Account">
+  <icon-account />
+</a>
+```
+
+[ꜛ Back to TOC](#table-of-contents)
+
+## Alt Text
+* Alt text is extremely useful for ensuring that images on the site can be used by people with various disabilities.
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
* Alt text is extremely useful for ensuring that images on the site can be used by people with various disabilities
```

File: accessibility/README.md [hunk 23]
```diff
@@ -0,0 +1,495 @@
...
+* Images can be categorised in different ways and there are some different recommendations for alt text depending on the type of image.
+* The W3C Web Accessibility Initiative has some good resources around Images:
+  * [Images Tutorial](https://www.w3.org/WAI/tutorials/images/) - Outlines the different purposes an image might have and recommendations for adding accessible text
+  * [`alt` Decision Tree](https://www.w3.org/WAI/tutorials/images/decision-tree/) - Provides a helpful decision tree on how to handle different situations with images, including when an empty `alt` attribute is appropriate.
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
  * [`alt` Decision Tree](https://www.w3.org/WAI/tutorials/images/decision-tree/) - Provides a helpful decision tree on how to handle different situations with images, including when an empty `alt` attribute is appropriate
```

File: accessibility/README.md [hunk 24]
```diff
@@ -0,0 +1,495 @@
...
+
+[ꜛ Back to TOC](#table-of-contents)
+
+## Element IDs
+
+* A variety of accessibility code and functionality (such as the `label` element's `for` attribute, `aria-labelledby` and other `aria-` attributes) use the `id` attribute to reference other elements
+  * For example, linking a checkbox input with a label using `id`/`for` allows the user to click on the label to toggle the checkbox
+  * This helps make the input easier to trigger with a mouse, and also makes it possible to style custom inputs (for example custom checkboxes) by applying styles to the `<label>` element
+* When writing components that use the `id` attribute - ensure that the `id` will always be unique
+  * Avoid hard coding an id for an element unless you know it will only be used once
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
  * Avoid hard coding an `id` for an element unless you know it will only be used once
```
Consistency.

File: accessibility/README.md [hunk 25]
```diff
@@ -0,0 +1,495 @@
...
+  * This is particularly important when the component is designed to be re-used, for example a swatch or a line item component
+  * Unique ids can be generated using passed in data, and additionally namespaces if needed
+  * For section type Canvas components - you could also pass the Shopify `section.id` as a prop and use that in your element id
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
  * For section type Canvas components - you could also pass the Shopify `section.id` as a prop and use that in your element `id`
```

File: accessibility/README.md [hunk 26]
```diff
@@ -0,0 +1,495 @@
...
+
+## Element IDs
+
+* A variety of accessibility code and functionality (such as the `label` element's `for` attribute, `aria-labelledby` and other `aria-` attributes) use the `id` attribute to reference other elements
+  * For example, linking a checkbox input with a label using `id`/`for` allows the user to click on the label to toggle the checkbox
+  * This helps make the input easier to trigger with a mouse, and also makes it possible to style custom inputs (for example custom checkboxes) by applying styles to the `<label>` element
+* When writing components that use the `id` attribute - ensure that the `id` will always be unique
+  * Avoid hard coding an id for an element unless you know it will only be used once
+  * This is particularly important when the component is designed to be re-used, for example a swatch or a line item component
+  * Unique ids can be generated using passed in data, and additionally namespaces if needed
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
  * Unique IDs can be generated using passed in data, and additionally namespaces if needed
```
This line is more referring to the concept of 'IDs' rather than the attribute `id`, hence the different casing.

File: accessibility/README.md [hunk 27]
```diff
@@ -0,0 +1,495 @@
...
+  * For section type Canvas components - you could also pass the Shopify `section.id` as a prop and use that in your element id
+
+Example of a dynamic id based on unique prop data:
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
Example of a dynamic `id` based on unique prop data: 
```

File: accessibility/README.md [hunk 28]
```diff
@@ -0,0 +1,495 @@
...
+      Quantity
+    </label>
+
+    <input
+      :id="`quantity-selector-input-${lineItem.key}`"
+      type="number"
+      value="1" 
+    >
+  </div>
+</template>
```
Comment (craigsbaldwin) [inline accessibility/README.md:456]: Add newline after `<template>` tag.

File: accessibility/README.md [hunk 29]
```diff
@@ -0,0 +1,495 @@
...
+<script>
+export default {
+  name: 'QuantitySelector',
+  
+  props: {
+
+    // Cart line item data from Shopify
```
Comment (craigsbaldwin) [inline accessibility/README.md:463]: Shouldn't have a newline before it as it's an inline comment. Only block comments require them.

File: accessibility/README.md [hunk 30]
```diff
@@ -0,0 +1,495 @@
...
+  name: 'VariantDropdown',
+  
+  props: {
+    namespace: {
+      type: String,
+      default: 'default'
+      required: true,
+    },
+    
+    options: {
```
Comment (craigsbaldwin) [inline accessibility/README.md:501]: We haven't decided yet on whether we'll have newlines between props yet.

File: accessibility/README.md [hunk 31]
```diff
@@ -0,0 +1,495 @@
...
+      class="visually-hidden"
+    >
+      Quantity
+    </label>
+
+    <select :id="`${namespace}-variant-dropdown`">
+      <option v-for="option in options">
+    </select>
+  </div>
+</template>
```
Comment (craigsbaldwin) [inline accessibility/README.md:489]: Add newline after `<template>`.

File: accessibility/README.md [hunk 32]
```diff
@@ -0,0 +1,495 @@
...
+<!-- Outputs select with the id `product-form-variant-dropdown` -->
+
+<variant-dropdown namespace="sticky-product-form" :options="options" />
+<!-- Outputs select with the id `sticky-product-form-variant-dropdown` -->
+```
+* Use unique data from the component's props, or add a `namespace` prop, or use a combination of both to ensure that the `id` is always unique
+* When using a `namespace` prop, make the prop `required` to ensure that its not missed
+* Utilise the browser tools and Storybook Accessibility checks that detect if an `id` is not unique
+* Consider your loading states as well as your standard/active states
+  * Often when a component is in a loading state it might not have access prop data that could be used for a unique id - For example using a `product.id` in an element's id, but the product is still being fetched and not yet set
```
Comment (craigsbaldwin) [inline accessibility/README.md]:
```suggestion
  * Often when a component is in a loading state it might not have access prop data that could be used for a unique `id` - For example using a `product.id` in an element's `id`, but the product is still being fetched and not yet set
```

**Review by craigsbaldwin — CHANGES_REQUESTED** [review]
Being super nitpicky but that's the point of coding guidelines.

File: accessibility/README.md (hunk 20 above)
Comment (georgia-wmw) [reply accessibility/README.md:361]: Good shout @csbrightside 

[Here's the W3 wiki which mentions labelling sections](https://www.w3.org/WAI/GL/wiki/Using_HTML5_section_element) (under User Agent and Assistive Technology Support Notes) and [here's a blog post updated in 2022](https://www.scottohara.me/blog/2021/07/16/section.html) outlining the same thing (they mention `title` but then suggest against it so we're good to stick with `aria-labelledby` and `aria-label`).

File: accessibility/README.md (hunk 18 above)
Comment (georgia-wmw) [reply accessibility/README.md]: [W3 states that using `role="presentation"` isn't as widely supported as an empty `alt` attribute](https://www.w3.org/WAI/tutorials/images/decorative/), and using an empty `alt` is the first thing they recommend.

Depending on the image, it might be better to stick with empty `alt`s alone, just in case a future upload is informative too

File: accessibility/README.md (hunk 30 above)
Comment (jamespeilow) [reply accessibility/README.md:501]: Removed the newline for now

File: accessibility/README.md (hunk 20 above)
Comment (jamespeilow) [reply accessibility/README.md:361]: Should we consider removing `aria-labelledby` on all our `section` elements?

I have been looking through the documentation that @georgia-wmw linked, along with adding some further reading and testing and I can't see that labelling all `section` elements is _required_ - it seems more like the opposite, that `section` elements should only be labelled in certain circumstances:

From the [Scott O'Hara blog post](https://www.scottohara.me/blog/2021/07/16/section.html#use-with-purpose):
> Regarding the `<section>` element’s implicit ARIA role, it is best named – thus exposing its role – only when absolutely necessary. Overpopulating a web page with landmarks will reduce their ability to help users find the most important parts of web pages. In other words, if everything’s a landmark then there’s nothing really special about them, is there? Instead, rely on proper heading levels to indicate implicit sections and sub-sections of your content.  

From [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role#accessibility_concerns):
> Use sparingly! [Landmark roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#3._landmark_roles) are intended to be used sparingly, to identify larger overall sections of the document. Using too many landmark roles can create "noise" in screen readers, making it difficult to understand the overall layout of the page.

I also did a basic test of some `<section>` elements without `aria-labelledby` on this codepen - https://codepen.io/jamespeilow/pen/vYvEeXN 
I did not see any accessibility issues flagged by Lighthouse or aXe Dev Tools, nor could I find any related rules in the [aXe Dev tools List of Rules](https://dequeuniversity.com/rules/axe/4.7)

I also did some testing with Voiceover on Mac on some of our recently launched Canvas sites, and there's a good amount of sites where the element referenced by `aria-labelledby` does not exist.
Given that the default ID for the `aria-labelledby` attribute that the Canvas `component` command creates contains the `section.id` - this results in Voiceover reading out the `id` string - for example: `section-title-template--16778842603746__5f8cbe58-204b-4065-b774-63f69e728aaf`. (Seen on [Stephen Webster](https://www.stephenwebster.com/))

If using `section` with `aria-labelledby` is best practice, I'm happy to keep this in Canvas, update this documentation and work on developer training to make sure its used properly and not ignored - but I couldn't find anything to back up using it on every section.

Also aware this could be broken out to a different discussion/ticket as it affects core Canvas.

**tldr: It looks like it might not be recommended to have `aria-labelledby`/`aria-label` on every `section` element, so it might be worth reviewing our usage of it**

File: accessibility/README.md (hunk 18 above)
Comment (jamespeilow) [reply accessibility/README.md]: I think there's no action here? The _"empty `alt` with `visually-hidden` text"_ described here applies to interactive elements (buttons/links etc) and there is mention of empty alt texts in the Alt Text section for standard images

File: accessibility/README.md (hunk 10 above)
Comment (jamespeilow) [reply accessibility/README.md:170]: @csbrightside - I added this example in as a response to seeing this sort of `click` handler around an `input` element in some PLP filter implementations (I can see things like this on both Watsons and HyperX) - so the main purpose of this example was to address that. 

I agree that `v-model` would be the ideal for working with inputs but I'm not sure how that would look when working with a third party filter library like Algolia, and I think in the context of filters, using `v-model` can get complicated when you have a number of input elements. 

I'll do some investigation on `v-model` and see if I can create a suitable example for that use case of filtering on a PLP.

In the meantime do you think its OK to keep this example as is?

File: accessibility/README.md (hunk 20 above)
Comment (georgia-wmw) [reply accessibility/README.md:361]: Stephen Webster is like a weird fluke of a site where its lighthouse score is/was 99-100 but when you actually interact using keyboard-only/a screenreader it becomes ~~a nightmarish hellscape~~ tricky to navigate 😵‍💫 Anyways

You're correct in that too many landmarks can get confusing for a webpage. I did some digging and [technique ARIA13](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA13) recommends using `aria-labelledby` for landmarks & regions, but then [technique ARIA20](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA20) states 'don't use too many landmarks'. So the instructions seem a bit out of order there. [MDN also states to use `region`s with labels sparingly](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role#accessibility_concerns) so that aXe flag could have been deprecated.

We could try and control landmark labels from CANVAS instead of passing this to every component. What if we only explicitly label PDPs and PLPs? I think those are pretty standard and predictable parts of all/most client websites. Then we can rely on appropriate HTML headings for segmenting the content itself (conscious that this is often up to client/designs and not always something in our control). The `aria-labelledby`s could use `{{ product.title }}` and `{{ collection.title }}` by default too, and there should be a loading state for both of these elements, so the ID would be present straight from page load (though that might get a bit repetitive)

Comment (jamespeilow) [reply accessibility/README.md:361]: Thanks for that ARIA13 link @georgia-wmw.
The one thing that isn't clear with that is its relation to the `<section>` element - ARIA13 describes that "landmarks and regions" should use `aria-labelledby` - However a `<section>` element seems like a bit of a special case in that it only becomes a `region`/gets an implied `role="region"` when it is given an accessible name:

[From MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role#description)
> Using the [`<section>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section) element will automatically communicate a section has a role of region if it is given an accessible name. Developers should always prefer using the correct semantic HTML element, in this case <section>, over using ARIA.

I do think we could consider still using `aria-labelledby` for some of the key default Canvas components, but then removing it from the component template for other section type components - a developer could always add it back to a certain section if needed.

Loading state is probably another thing to consider - If a section is loading and the `aria-labelledby` element depends on data that isn't ready, I wonder if having "loading" as the `aria-labelledby`text is better or if its better to have nothing, or some other way of handling that

File: accessibility/README.md (hunk 10 above)
Comment (craigsbaldwin) [reply accessibility/README.md:170]: Yeah it's good.

**Review by craigsbaldwin — APPROVED** [review]
I actioned the remaining nitpicks to save the back and forth.

Approved as it is, just need a decision on section labels.

File: accessibility/README.md (hunk 20 above)
Comment (craigsbaldwin) [reply accessibility/README.md:361]: Do we just keep the labels on the sections that use the `template-title` ID instead of the `section-title-{{ section.id }}`? The assumption being those are on the 'main' components which are important to the page and not used on section everywhere sections.

Comment (georgia-wmw) [reply accessibility/README.md:361]: I think that's a good idea! It's an easily identifiable landmark & most/all instances of `template-title` on the canvas repo reference a `h1` so there should only be 1 per page (keeping them sparse).

EDIT: I think we can ignore loading states for now. Anecdotally I've noticed that if an id isn't present in the Liquid loading state of a Vue component then Lighthouse would flag the 'id not found' error but if we're removing `aria-labelledby`s from sections then this wouldn't matter. I don't think we need a separate 'loading' string either, because the content of the loading states should be using `loading` strings (at least that's what I've used in Stephen Webster & Childsplay).

BUT [there's an aria attribute called `aria-live`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) which covers dynamically updating portions of a webpage without reloading the page that caught my eye a while back which looks really interesting! Conscious that this is probably a conversation for another day, but its something we could keep in mind.

**Comment by jamespeilow** [comment]
@csbrightside - it looks like your in-situ code review commit has undone some of the changes that I addressed in my previous 2 commits - was that intentional? If not could you re-check and resolve those issues?

File: accessibility/README.md (hunk 20 above)
Comment (jamespeilow) [reply accessibility/README.md:361]: I agree - Keeping the labels just for `template-title` ID sounds like a good approach considering they are the "main" sections/components on the site.

**Review by craigsbaldwin — APPROVED** [review]
Oops sorry, wasn't intentional, reverted.

File: accessibility/README.md (hunk 20 above)
Comment (craigsbaldwin) [reply accessibility/README.md:361]: @jamespeilow Excellent, I'll look at updating the `component` command and removing labels from non-template ID sections, do you want to flag which sections should be labelled?

Comment (jamespeilow) [reply accessibility/README.md:361]: @csbrightside @georgia-wmw - added a new section to call out using `aria-labelledby` with `<section>` elements, along with a general point in the `aria-labelledby` section to say that it probably doesn't need using in a lot of cases

File: accessibility/README.md [hunk 33]
```diff
@@ -0,0 +1,542 @@
...
+  </h1>
+
+  <!-- Other content -->
+</section>
+```
+
+* In many cases, elements get accessible names from either their inner content (e.g. `<button>`, `<a>` or `<td>` elements), or from associated content (e.g. an `<input>` and it's `<label>` element), so there are many cases you won't need to reach for `aria-labelledby`
+* Ensure that the element referenced by `aria-labelledby` exists on the page
+  * If the referenced element is missing, screen readers will still read out the the other text content contained in the element, but users may be missing out on the additional context that the referenced element might be adding
+* When using `aria-labelledby` in a component, consider whether the element you are referencing existing inside the component's template code
```
Comment (georgia-wmw) [inline accessibility/README.md]: Looks like a small typo here, should this read
```
When using `aria-labelledby` in a component, consider whether the element you are referencing exists inside the component's template code
```

**Review by craigsbaldwin — APPROVED** [review]
Looks good, approved.

File: accessibility/README.md [hunk 34]
```diff
@@ -0,0 +1,542 @@
...
+  aria-label="Account"
+>
+  <icon-account />
+</a>
+```
+
+### Ensure correct usage of `aria-labelledby` on `<section>` elements
+
+* When a `<section>` element is given an accessible name, for example with the `aria-labelledby` or `aria-label` attribute, the element receives an implicit ARIA attribute [`role="region"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/region_role) which marks it as a [landmark](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#3._landmark_roles) for screen readers and other assistive technologies
+* These landmarks can then be used as another method for screen reader users to navigate around the page and understand it's contents
```
Comment (georgia-wmw) [inline accessibility/README.md]: One small typo here - `it's` should be `its`
```These landmarks can then be used as another method for screen reader users to navigate around the page and understand its contents```

File: accessibility/README.md (hunk 20 above)
Comment (georgia-wmw) [reply accessibility/README.md:361]: @jamespeilow Looks good! I like the way everything's phrased. Spotted a couple of typos but otherwise it looks good from my end

**Review by craigsbaldwin — APPROVED** [review]

**Review by georgia-wmw — APPROVED** [review]
